doc/vxlan/EVPN/EVPN_VxLAN_Multihoming.md

```diff
@@ -0,0 +1,1821 @@
+; Producer: fpmsyncd
+; Consumer: fdborch
+; Status: stable
+key               = EVPN_ES_BACKUP_NHG_TABLE:"ifname"if_id
+						; vlan_id is 1-4 DIGIT VLAN ID.
+						; if_id is interface number
+; field = value
+nexthop_group        = nexthop_group_value
+            ; Backup nexthop group id for the given ES
+						; Singular value.
+
+```
+
+**VXLAN_FDB_TABLE**
+
+Producer:  Fdbsyncd
+
+Consumer: FdbOrch
+
+Description: Updated existing table to store next-hop group.
+
+Schema:
+
+```
+; Existing table
+; New field is added for L2 ECMP support
+
+key = VXLAN_FDB_TABLE:"Vlan"vlanid:mac_address
+                          ; MAC Address and VLAN ID
+; field = value
+remote_vtep   = IPv4/IPv6 address           (existing field)
+nexthop_group = L2_NEXTHOP_GROUP_TABLE:key (new field)
+				; index within the L2_NEXTHOP_GROUP_TABLE
+				;
+				used instead of remote_vtep field
+vni           = 1*8DIGIT                    (existing field)
+ifname        = STRING                      (new field)
+				; Local interface name in case
+				; mac is received with local ESI.
+type          = "dynamic"/"static/dynamic_control_learn"  (existing field but new type)
+```
+
+
+**L2_NEXTHOP_GROUP_TABLE**
+
+Producer:  Fdbsyncd
+
+Consumer: L2NhgOrch
+
+Description: L2 Next-Hop Group Table entry
+
+Schema:
+
+```
+; New table
+; Updated group to identify split horizon group the Next-hop group member belongs to.
+
+key = L2_NEXTHOP_GROUP_TABLE:nhid
+                  ; Nexthop group id
+; field = value
+remote_vtep   = IPv4/IPv6 address
+nexthop_group = L2_NEXTHOP_GROUP_TABLE:key
+				; index within the L2_NEXTHOP_GROUP_TABLE separated
+				; by "," used for recursive/ECMP routes.
+				; ( When this field is present, other fields will
+				; not be present)
+```
+
+<a id="323-STATE-DB"></a>
+### 3.2.3 STATE DB
+
+No changes
+
+<a id="324-ASIC-DB"></a>
+### 3.2.4 ASIC DB
+
+Refer to Section 3.4 for the new SAI objects and changes to existing SAI objects.
+The ASIC DB changes will reflect the above.
+
+#<a id="325-COUNTER-DB"></a>
+## 3.2.5 COUNTER DB
+
+No Changes
+
+#<a id="33-Impact-To-Existing-Modules"></a>
+# 3.3 Impact to existing modules
+
+The following components will undergo changes to support EVPN Multihoming:
+
+- TeamMgrd
+- Zebra
+- Fpmsyncd
+- Fdbsyncd
+- Fdborch
+- VxlanOrch
+
+No impact is anticipated to MCLAG and LAG modules, e.g.
+
+- ICCPd
+- mclagsyncd
+
+<a id="331-MCLAG-Management-Framework"></a>
+### 3.3.1 MCLAG Management Framework
+Configuration handling of MCLAG should now be aware of EVPN Ethernet-Segment configurations and throw appropriate errors in order to achieve mutual-exclusion between EVPN Multihoming and MCLAG features.
+OpenConfig evpn yang is imported into SONiC and ethernet-segment URIs are implemented.
+OpenConfig LACP yang already has system-id-mac configuration and it will be implemented.
+OpenConfig evpn yang will be extended for global non-standard parameter configurations (e.g. MAC/ARP hold-timer, etc.)
```

Comment:
We dont have openconfig support for EVPN feature as such in the SONiC community, are we planning to bring that support via mgmt-framework as part of EVPN MH effort?

Reply:
@venkatmahalingam , yes, that's the plan.